lib/charms/layer/apache_kafka.py

```diff
@@ -135,3 +152,48 @@ def stop(self):
     def cleanup(self):
         self.dist_config.remove_users()
         self.dist_config.remove_dirs()
+
+    def get_ip_for_interface(self, network_interface):
+        """
+        Helper to return the ip address of this machine on a specific
+        interface.
+
+        @param str network_interface: either the name of the
+        interface, or a CIDR range, in which we expect the interface's
+        ip to fall. Also accepts 0.0.0.0 (and variants, like 0/0) as a
+        special case, which will simply return what you passed in.
+
+        """
+        if network_interface.startswith('0') or network_interface == '::':
+            # Allow users to reset the charm to listening on any
+            # interface.  Allow operators to specify this however they
+            # wish (0.0.0.0, ::, 0/0, etc.).
+            return network_interface
+
+        # Is this a CIDR range, or an interface name?
+        is_cidr = len(network_interface.split(".")) == 4 or len(
+            network_interface.split(":")) == 8
+
+        if is_cidr:
+            interfaces = netifaces.interfaces()
+            for interface in interfaces:
+                try:
+                    ip = netifaces.ifaddresses(interface)[2][0]['addr']
```

Comment:
Instead of hard coding [2] here, netifaces suggests using `netifaces.AF_INET` to get the ipv4 entry:

https://pypi.python.org/pypi/netifaces

We'll want to handle ipv6 too, so maybe try getting the ip4 addr here, then try for ip6 (`netifaces.AF_INET6`), and then except your fate .